modcc/printer/cprinter.cpp

```diff
@@ -278,208 +319,141 @@ std::string emit_cpp_source(const Module& module_, const printer_options& opt) {
             out << ";\n";
         }
     }
-    out << "\n";
-
-    out << "// interface methods\n";
-    out << "void init(" << ppack_name << "* pp) {\n" << indent;
+    out << "\n"
+        << "// interface methods\n";
+    out << "static void init(arb_mechanism_ppack* pp) {\n" << indent;
     emit_body(init_api);
+    if (init_api && init_api->body() && !init_api->body()->statements().empty()) {
+        auto n = std::count_if(vars.arrays.begin(), vars.arrays.end(),
+                               [] (const auto& v) { return v->is_state(); });
+        out << fmt::format(FMT_COMPILE("if (!{0}multiplicity) return;\n"
+                                       "for (arb_size_type ix = 0; ix < {1}; ++ix) {{\n"
```

Comment:
I'll add escaping curly braces to the 'con' column for using `fmt` to write generated C++ code.